test/fetch_multi_test.rb

```diff
@@ -223,6 +223,49 @@ def test_fetch_multi_on_derived_model_raises
     end
   end
 
+  def test_fetch_multi_with_mixed_hits_and_misses_returns_only_readonly_records
+    IdentityCache.with_fetch_read_only_records do
+      cache_response = {}
+      cache_response[@bob_blob_key] = cache_response_for(@bob)
+      cache_response[@joe_blob_key] = nil
+      cache_response[@fred_blob_key] = cache_response_for(@fred)
+      fetch_multi = fetch_multi_stub(cache_response)
+      assert_equal [@bob, @joe, @fred], Item.fetch_multi(@bob.id, @joe.id, @fred.id)
+      assert fetch_multi.has_been_called_with?(@bob_blob_key, @joe_blob_key, @fred_blob_key)
+
+      Item.fetch_multi(@bob.id, @joe.id, @fred.id).each do |record|
```

Comment:
While I understand that `fetch_multi_stub` means we're not actually caching anything, calling `Item.fetch_multi` twice in the testcase is confusing. Naively, I'd expect the first call to cache the 3 records (a mix of hits and misses) and the second to return the cached records (all hits). Can you instead do something like:
```ruby
response = Item.fetch_multi(@bob.id, @joe.id, @fred.id)
assert_equal [@bob, @joe, @fred], response
...
response.each do |record|
...
```